lib/scope.js

```diff
@@ -119,38 +119,26 @@ Scope.prototype.delete = function _delete(uri, requestBody, options) {
 };
 
 Scope.prototype.pendingMocks = function pendingMocks() {
-  return Object.keys(this.keyedInterceptors);
+  var self = this;
+
+  var pendingInterceptorKeys = Object.keys(this.keyedInterceptors).filter(function (key) {
+    var interceptorList = self.keyedInterceptors[key];
+    var pendingInterceptors = interceptorList.filter(function (interceptor) {
+      var persistedAndUsed = self._persist && interceptor.interceptionCounter > 0;
+      return !persistedAndUsed && !interceptor.optional;
```

Comment:
This piece of code (and the one it replaced) is hiding an assumption: _when an interceptor has been fully consumed, it's removed from the `keyedInterceptors` list_.

I would like to get rid of that assumption. This however would require a refactor that touches various parts of the codebase. In the spirit of keeping this PR on target, I would instead suggest adding a comment to document the fact and I'll add an issue to refactor it.